### PR TITLE
Dm 1175 and Dm 1466

### PR DESCRIPTION
**Pull request by JoshingYou1**
### JIRA issue links
https://agile6.atlassian.net/browse/DM-1175
https://agile6.atlassian.net/browse/DM-1466

## Description - what does this code do?
Adds new landing pages for the Shark Tank and GoFish! competitions. 
Adds iNET partner page.

## Testing done - how did you test it/steps on how can another person can test it
**Run the following items in the terminal:**
1. `rails db:migrate`
2. `rails go_fish_practices:assign_go_fish_badge`
3. `rails shark_tank_practices:assign_shark_tank_badge`
4. `rails inet_partner_practices:assign_inet_partner`

Log in as any user type and browse to both `/competitions/go-fish` and `/competitions/shark-tank`, or just use the links in the header drop-down labelled "Bulletin."

On both pages, click on one of the practice titles or main display image and make sure it takes you to that practice's show page. You can also favorite a practice.

For the Shark Tank Page, the "Apply now" button doesn't function yet(waiting for the survey monkey link). For the GoFish! page, the "FAQ" link only works if you have a VPN. All other links should function properly.

Visit `/partners` and find the "VHA Innovators Network (INET)" partner. Click on it and make sure it takes you to the "iNET" partner show page. Once there, you should see the practices that have iNET as one of their partners. To test this out, you can go to the Overview page of the practice editor(if logged in as an admin/approver) and check/uncheck the "VHA Innovators Network" option. If you go back to the iNET page, that change should be reflected.

## Screenshots, Gifs, Videos from application (if applicable)
Screen Shot 2020-04-14 at 3 46 14 PM
Screen Shot 2020-04-14 at 3 46 57 PM
Screen Shot 2020-04-14 at 3 47 30 PM
Screen Shot 2020-04-14 at 3 47 56 PM
Screen Shot 2020-04-14 at 3 48 45 PM

## Link to mock-ups/mock ups (image file if you have it) (if applicable)
https://www.figma.com/file/DDUSxYiD1MaWNrGFmMjY8d8a/Diffusion-Marketplace-Designs?node-id=4024%3A0

## Acceptance criteria
**DM-1175:**

Large call to action button (e.g. Apply to Shark Tank) at center stage above the fold

Single landing page that is simple yet visually appealing (marketplace.va.gov/sharktank2020)

The  call to action would lead to the Survey Monkey application being managed by the Atlas team (Rachel and Arielle, copied)

The ability to post several information documents and/or links such as Shark Tank Information Session pdf, FAQ sheet, Youtube link to 2019 Shark Tank, etc

Nice to haves:
It could be cool to have some tiles below the fold linking to last year’s Shark Tank winners.

I’ll defer to the team if you have any recommendations, such as embedding the survey in the marketplace for a more seamless experience.

Re-usability:  If we get this right then it could be re-purposed for things like Spark Seed Spread by Innovators Network, and in future years.

Add a link in the top nav menu

**DM-1466:**

Design a basic landing page similar to Shark Tank

Have copy at top and “iNET Practices” below

Add “Go Fish” to header like Shark Tank, takes you to page

Dev: Follow designs

Use unique URL

Make sure all the links are active

Dev: Create an “iNET Partners” group 

Publish iNET on “Partners Page”

Only CTAs are links in docs

Write unit tests 

## Definition of done
- [X] Unit tests written (if applicable)
- [X] e2e/accessibility tests written (if applicable)
- [X] Events are logged appropriately
- [X] Documentation has been updated, if applicable
- [X] A link has been provided to the originating JIRA issue
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: db/migrate/20200409155443_add_inet_partner_to_practice_partners.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+class AddInetPartnerToPracticePartners < ActiveRecord::Migration[5.2]
+  def change
+    iNet = PracticePartner.find_by(slug: 'vha-innovators-network')
+    unless iNet
+      PracticePartner.create!({name: 'VHA Innovators Network', short_name: 'iNET', description: 'The VHA Innovators Network (iNET) is a network of 33 VA medical centers changing the way employees think and solve problems through training and accelerated operationalizing innovation.', slug: 'vha-innovators-network'})
```
Comment (emyl3) [inline db/migrate/20200409155443_add_inet_partner_to_practice_partners.rb:5]: Running `rake db:migrate` was throwing errors for me:
```
rake aborted!
StandardError: An error has occurred, this and all later migrations canceled:

undefined method `id' for nil:NilClass
/Users/elisa/skylight/diffusion-marketplace/db/migrate/20200409155443_add_inet_partner_to_practice_partners.rb:21:in `block in change'
/Users/elisa/skylight/diffusion-marketplace/db/migrate/20200409155443_add_inet_partner_to_practice_partners.rb:17:in `each'
/Users/elisa/skylight/diffusion-marketplace/db/migrate/20200409155443_add_inet_partner_to_practice_partners.rb:17:in `change'

Caused by:
NoMethodError: undefined method `id' for nil:NilClass
/Users/elisa/skylight/diffusion-marketplace/db/migrate/20200409155443_add_inet_partner_to_practice_partners.rb:21:in `block in change'
/Users/elisa/skylight/diffusion-marketplace/db/migrate/20200409155443_add_inet_partner_to_practice_partners.rb:17:in `each'
/Users/elisa/skylight/diffusion-marketplace/db/migrate/20200409155443_add_inet_partner_to_practice_partners.rb:17:in `change'
Tasks: TOP => db:migrate
(See full trace by running task with --trace)
```
setting `iNet = PracticePartner.create....` helped fix this issue for me...

However, spoke with @JoshingYou1 and Josh did not run into issue even though he also did not have a iNet PracticePartner in his DB... 🧐

Comment (ahay-agile6) [reply db/migrate/20200409155443_add_inet_partner_to_practice_partners.rb:5]: yes,
we need an `iNet =` on line 5 here.
thanks for catching that @emyl3.

Comment (JoshingYou1) [reply db/migrate/20200409155443_add_inet_partner_to_practice_partners.rb]: Updated! Thank you!

File: db/migrate/20200402174615_add_shark_tank_badge_to_badges.rb [hunk 2]
```diff
@@ -1,5 +1,44 @@
 class AddSharkTankBadgeToBadges < ActiveRecord::Migration[5.2]
   def change
-    Badge.create({name: 'Shark Tank'})
+    Badge.find_or_create_by!({name: 'Shark Tank'})
```
Comment (ahay-agile6) [inline db/migrate/20200402174615_add_shark_tank_badge_to_badges.rb:3]: you can set this to 
`shark_tank_badge = Badge.find_or_create_by!({name: 'Shark Tank'})`

File: db/migrate/20200402174615_add_shark_tank_badge_to_badges.rb [hunk 3]
```diff
@@ -1,5 +1,44 @@
...
+
+    shark_tank_badge_id = Badge.find_by(name: 'Shark Tank').id
```
Comment (ahay-agile6) [inline db/migrate/20200402174615_add_shark_tank_badge_to_badges.rb:5]: then get rid of this

File: db/migrate/20200402174615_add_shark_tank_badge_to_badges.rb [hunk 4]
```diff
@@ -1,5 +1,44 @@
...
+        'national-telewound-care-practice',
+        'the-stride-program',
+        'my-life-my-story',
+        'vione',
+        'flow3'
+    ]
+    shark_tank_winner_slugs.each do |stws|
+      shark_tank_practice = Practice.find_by(slug: stws)
+      if !shark_tank_practice.nil?
+        BadgePractice.create!({practice_id: shark_tank_practice.id, badge_id: shark_tank_badge_id})
```
Comment (ahay-agile6) [inline db/migrate/20200402174615_add_shark_tank_badge_to_badges.rb:38]: then do this:
`BadgePractice.create!({practice: shark_tank_practice, badge: shark_tank_badge})`

File: db/migrate/20200413182933_add_go_fish_badge_to_badges_and_create_go_fish_badge_practices.rb [hunk 5]
```diff
@@ -1,6 +1,6 @@
 class AddGoFishBadgeToBadgesAndCreateGoFishBadgePractices < ActiveRecord::Migration[5.2]
   def change
-    Badge.create({name: 'Go Fish'})
+    Badge.find_or_create_by!({name: 'Go Fish'})
```
Comment (ahay-agile6) [inline db/migrate/20200413182933_add_go_fish_badge_to_badges_and_create_go_fish_badge_practices.rb:3]: same thing here

File: db/migrate/20200413182933_add_go_fish_badge_to_badges_and_create_go_fish_badge_practices.rb [hunk 6]
```diff
@@ -1,6 +1,6 @@
...
 
     go_fish_badge_id = Badge.find_by(name: 'Go Fish').id
```
Comment (ahay-agile6) [inline db/migrate/20200413182933_add_go_fish_badge_to_badges_and_create_go_fish_badge_practices.rb:5]: and here, get rid of this

File: db/migrate/20200413182933_add_go_fish_badge_to_badges_and_create_go_fish_badge_practices.rb [hunk 7]
```diff
@@ -35,7 +35,7 @@ def change
     go_fish_practice_slugs.each do |gfps|
       go_fish_practice = Practice.find_by(slug: gfps)
       if !go_fish_practice.nil?
-        BadgePractice.create({practice_id: go_fish_practice.id, badge_id: go_fish_badge_id})
+        BadgePractice.create!({practice_id: go_fish_practice.id, badge_id: go_fish_badge_id})
```
Comment (ahay-agile6) [inline db/migrate/20200413182933_add_go_fish_badge_to_badges_and_create_go_fish_badge_practices.rb:38]: then use `practice` and `badge` with their respective variables

**Comment by JoshingYou1** [comment]
I think I got everything. Let me know if I didn't please.

**Pull request merged**